scripts/artifacts-building/apt/build-apt-artifacts.sh

```diff
@@ -74,10 +74,37 @@ fi
 # from previously installed/configured items.
 rm -rf /etc/ansible /etc/openstack_deploy /usr/local/bin/ansible* /usr/local/bin/openstack-ansible*
 
-# Install Ansible
-./scripts/bootstrap-ansible.sh
+# Bootstrap Ansible
+# This script is sourced to ensure that the common
+# functions and vars are available.
+source scripts/bootstrap-ansible.sh
 cp scripts/artifacts-building/apt/lookup/* /etc/ansible/roles/plugins/lookup/
 
+# Figure out when it is safe to automatically replace artifacts
```

Comment:
Why not put this whole repeated block inside safe_to_replace_artifacts? Or some other function?

Reply:
That would make it very abstracted IMO, while this is easily readable.

Reply:
@hughsaunders The reason is that line 81 is different in each set of artifact scripts.

Reply:
Ahh, apologies, didn't spot that.